js/src/admin/components/UserListPage.tsx

```diff
@@ -0,0 +1,393 @@
+                      'title',
+                      extractText(app.translator.trans('core.admin.user_list.grid.default_columns.email_visibility_hide'))
+                    );
+
+                    // Replace placeholder email with real email
+                    emailAddress.text(user.email());
+
+                    // Change button icons
+                    emailToggleButtonIcon.addClass('fa-eye');
+                    emailToggleButtonIcon.removeClass('fa-eye-slash');
+                  }
+
+                  emailContainer.attr('data-email-shown', !emailShown);
+                }}
+                class="UserList-emailIconBtn"
+                title={app.translator.trans('core.admin.user_list.grid.default_columns.email_show')}
+              >
+                {icon('far fa-eye-slash fa-fw', { className: 'icon' })}
+              </button>
+            </div>
+          );
+        },
+      },
+      70
+    );
+
+    columns.add(
+      'editUser',
+      {
+        name: app.translator.trans('core.admin.user_list.grid.default_columns.edit_user'),
+        content: (user: User) => (
+          <Button
+            className="Button UserList-editModalBtn"
+            title={app.translator.trans('core.admin.user_list.grid.default_columns.edit_user_tooltip', { username: user.username() })}
+            onclick={() => app.modal.show(EditUserModal, { user })}
+          >
+            {app.translator.trans('core.admin.user_list.grid.default_columns.edit_user_button')}
+          </Button>
+        ),
+      },
+      -90
+    );
+
+    columns.add(
+      'profileLink',
+      {
+        name: app.translator.trans('core.admin.user_list.grid.default_columns.profile_link'),
+        content: (user: User) => {
+          const profileUrl = `${app.forum.attribute('baseUrl')}/u/${user.slug()}`;
+
+          return (
+            <a
+              class="UserList-profileLink"
+              target="_blank"
+              href={profileUrl}
+              title={extractText(
+                app.translator.trans('core.admin.user_list.grid.default_columns.profile_link_tooltip', { username: user.username() })
+              )}
+            >
+              {icon('fas fa-link')}
+            </a>
+          );
+        },
+      },
+      // This should probably come last unless an ext really wants to be last
+      -100
+    );
+
+    return columns;
+  }
+
+  nextPage() {
+    this.isLoadingPage = true;
+    // Make sure the render doesn't end up happening after the API refresh
+    m.redraw.sync();
+
+    const newPageNum = this.pageNumber + 1;
+    this.loadNextUserSet(newPageNum * this.numPerPage, newPageNum);
+  }
+
+  previousPage() {
+    this.isLoadingPage = true;
+    // Make sure the render doesn't end up happening after the API refresh
+    m.redraw.sync();
+
+    const newPageNum = this.pageNumber - 1;
+    this.loadNextUserSet(newPageNum * this.numPerPage, newPageNum);
+  }
+
+  /**
+   * Component to render.
+   */
+  content() {
```

Comment:
I think I'd prefer for this to be the very first method; we don't follow this perfectly, but so far there's a trend of:

- Mithril lifecycle hooks
- `view` (or equivalent)
- ItemList methods
- Functional methods